instrumenter.go

```diff
@@ -130,17 +130,22 @@ func (i *instrumenter) markConds(n ast.Node) bool {
 	case *ast.UnaryExpr:
 		if n.Op == token.NOT {
 			delete(i.marked, n)
-			i.marked[n.X] = true
+			i.marked[n] = true
+			return false
```

Comment:
> The function markConds should always return true, just in case the !, && or || expression contains a nested function body. I guess I didn't include such a case in my test suite.

Yes I missed this case. I will include this without returning false from `markConds`. Thank you.

> My initial suggestion was to just remove the UnaryExpr and BinaryExpr cases. Didn't that work well enough?

I can get as we expected for conditional statements
```
if i > 0 && s == "positive" {
		return "yes, positive"
	}
```
but not detecting the conditional assignments when we remove the UnaryExpr and BinaryExpr cases.

```
return i > 0
lengthValidation := len("") > 0
_ = !!!a
```

Reply:
The question now becomes whether a statement like `return i > 0` *should* be instrumented in branch coverage mode. I don't think so, as this is precisely the difference between condition coverage and branch coverage.

Reply:
These are not branches. I got it now.. Thank you.